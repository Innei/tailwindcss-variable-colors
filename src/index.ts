import type { DefaultColors } from 'tailwindcss/types/generated/colors'

type VariableColors = {
  [key in keyof DefaultColors]: DefaultColors[key] extends `#${string}`
    ? `rgb(var(--tw-color-${key})/var(--tw-text-opacity))`
    : DefaultColors[key] extends string
    ? // has no uppercase letter
      DefaultColors[key] extends Lowercase<DefaultColors[key]>
      ? key
      : DefaultColors[key]
    : {
        [level in keyof DefaultColors[key]]: level extends string | number
          ? `rgb(var(--tw-color-${key}-${level})/var(--tw-text-opacity))`
          : never
      }
}

export function createVariableColors(colors: DefaultColors) {
  const variableColors = {} as any

  for (const [key, value] of Object.entries(colors)) {
    if (typeof value === 'string') {
      if (!value.startsWith('#')) {
        // color keyword
        variableColors[key] = value
      } else {
        // key color
        variableColors[
          key
        ] = `rgb(var(--tw-color-${key})/var(--tw-text-opacity))`
      }
    } else {
      // tonal palette
      variableColors[key] = {}
      for (const [level] of Object.entries(value)) {
        variableColors[key][
          level
        ] = `rgb(var(--tw-color-${key}-${level})/var(--tw-text-opacity))`
      }
    }
  }

  return variableColors as VariableColors
}
