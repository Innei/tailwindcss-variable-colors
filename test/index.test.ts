import colors from 'tailwindcss/colors'

import { describe, expect, it } from 'vitest'
import { createVariableColors } from '../src'

describe('should', () => {
  const variableColors = createVariableColors(colors)
  it('generate color keyword', () => {
    expect(variableColors.inherit).toBe(`inherit`)
  })
  it('generate color keyword (camelCase)', () => {
    expect(variableColors.current).toBe(`currentColor`)
  })
  it('generate key color', () => {
    expect(variableColors.white).toBe(
      `rgb(var(--tw-color-white)/var(--tw-text-opacity))`,
    )
  })
  it('generate tonal palette', () => {
    expect(variableColors.slate['50']).toBe(
      `rgb(var(--tw-color-slate-50)/var(--tw-text-opacity))`,
    )
  })
  it('generate tonal palette (deprecated)', () => {
    expect(variableColors.lightBlue['50']).toBe(
      `rgb(var(--tw-color-lightBlue-50)/var(--tw-text-opacity))`,
    )
  })
  it('match snapshot', () => {
    expect(variableColors).toMatchSnapshot()
  })
})
