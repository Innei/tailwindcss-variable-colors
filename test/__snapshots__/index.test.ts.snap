// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should > match snapshot 1`] = `
{
  "amber": {
    "100": "rgb(var(--tw-color-amber-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-amber-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-amber-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-amber-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-amber-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-amber-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-amber-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-amber-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-amber-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-amber-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-amber-950)/var(--tw-text-opacity))",
  },
  "black": "rgb(var(--tw-color-black)/var(--tw-text-opacity))",
  "blue": {
    "100": "rgb(var(--tw-color-blue-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-blue-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-blue-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-blue-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-blue-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-blue-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-blue-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-blue-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-blue-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-blue-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-blue-950)/var(--tw-text-opacity))",
  },
  "blueGray": {
    "100": "rgb(var(--tw-color-blueGray-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-blueGray-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-blueGray-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-blueGray-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-blueGray-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-blueGray-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-blueGray-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-blueGray-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-blueGray-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-blueGray-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-blueGray-950)/var(--tw-text-opacity))",
  },
  "coolGray": {
    "100": "rgb(var(--tw-color-coolGray-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-coolGray-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-coolGray-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-coolGray-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-coolGray-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-coolGray-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-coolGray-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-coolGray-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-coolGray-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-coolGray-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-coolGray-950)/var(--tw-text-opacity))",
  },
  "current": "currentColor",
  "cyan": {
    "100": "rgb(var(--tw-color-cyan-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-cyan-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-cyan-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-cyan-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-cyan-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-cyan-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-cyan-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-cyan-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-cyan-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-cyan-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-cyan-950)/var(--tw-text-opacity))",
  },
  "emerald": {
    "100": "rgb(var(--tw-color-emerald-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-emerald-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-emerald-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-emerald-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-emerald-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-emerald-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-emerald-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-emerald-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-emerald-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-emerald-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-emerald-950)/var(--tw-text-opacity))",
  },
  "fuchsia": {
    "100": "rgb(var(--tw-color-fuchsia-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-fuchsia-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-fuchsia-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-fuchsia-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-fuchsia-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-fuchsia-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-fuchsia-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-fuchsia-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-fuchsia-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-fuchsia-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-fuchsia-950)/var(--tw-text-opacity))",
  },
  "gray": {
    "100": "rgb(var(--tw-color-gray-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-gray-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-gray-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-gray-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-gray-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-gray-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-gray-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-gray-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-gray-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-gray-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-gray-950)/var(--tw-text-opacity))",
  },
  "green": {
    "100": "rgb(var(--tw-color-green-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-green-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-green-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-green-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-green-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-green-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-green-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-green-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-green-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-green-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-green-950)/var(--tw-text-opacity))",
  },
  "indigo": {
    "100": "rgb(var(--tw-color-indigo-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-indigo-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-indigo-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-indigo-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-indigo-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-indigo-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-indigo-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-indigo-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-indigo-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-indigo-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-indigo-950)/var(--tw-text-opacity))",
  },
  "inherit": "inherit",
  "lightBlue": {
    "100": "rgb(var(--tw-color-lightBlue-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-lightBlue-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-lightBlue-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-lightBlue-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-lightBlue-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-lightBlue-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-lightBlue-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-lightBlue-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-lightBlue-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-lightBlue-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-lightBlue-950)/var(--tw-text-opacity))",
  },
  "lime": {
    "100": "rgb(var(--tw-color-lime-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-lime-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-lime-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-lime-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-lime-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-lime-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-lime-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-lime-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-lime-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-lime-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-lime-950)/var(--tw-text-opacity))",
  },
  "neutral": {
    "100": "rgb(var(--tw-color-neutral-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-neutral-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-neutral-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-neutral-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-neutral-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-neutral-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-neutral-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-neutral-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-neutral-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-neutral-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-neutral-950)/var(--tw-text-opacity))",
  },
  "orange": {
    "100": "rgb(var(--tw-color-orange-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-orange-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-orange-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-orange-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-orange-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-orange-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-orange-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-orange-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-orange-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-orange-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-orange-950)/var(--tw-text-opacity))",
  },
  "pink": {
    "100": "rgb(var(--tw-color-pink-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-pink-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-pink-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-pink-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-pink-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-pink-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-pink-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-pink-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-pink-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-pink-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-pink-950)/var(--tw-text-opacity))",
  },
  "purple": {
    "100": "rgb(var(--tw-color-purple-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-purple-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-purple-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-purple-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-purple-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-purple-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-purple-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-purple-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-purple-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-purple-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-purple-950)/var(--tw-text-opacity))",
  },
  "red": {
    "100": "rgb(var(--tw-color-red-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-red-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-red-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-red-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-red-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-red-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-red-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-red-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-red-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-red-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-red-950)/var(--tw-text-opacity))",
  },
  "rose": {
    "100": "rgb(var(--tw-color-rose-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-rose-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-rose-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-rose-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-rose-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-rose-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-rose-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-rose-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-rose-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-rose-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-rose-950)/var(--tw-text-opacity))",
  },
  "sky": {
    "100": "rgb(var(--tw-color-sky-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-sky-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-sky-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-sky-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-sky-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-sky-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-sky-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-sky-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-sky-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-sky-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-sky-950)/var(--tw-text-opacity))",
  },
  "slate": {
    "100": "rgb(var(--tw-color-slate-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-slate-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-slate-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-slate-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-slate-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-slate-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-slate-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-slate-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-slate-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-slate-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-slate-950)/var(--tw-text-opacity))",
  },
  "stone": {
    "100": "rgb(var(--tw-color-stone-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-stone-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-stone-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-stone-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-stone-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-stone-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-stone-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-stone-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-stone-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-stone-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-stone-950)/var(--tw-text-opacity))",
  },
  "teal": {
    "100": "rgb(var(--tw-color-teal-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-teal-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-teal-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-teal-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-teal-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-teal-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-teal-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-teal-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-teal-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-teal-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-teal-950)/var(--tw-text-opacity))",
  },
  "transparent": "transparent",
  "trueGray": {
    "100": "rgb(var(--tw-color-trueGray-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-trueGray-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-trueGray-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-trueGray-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-trueGray-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-trueGray-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-trueGray-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-trueGray-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-trueGray-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-trueGray-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-trueGray-950)/var(--tw-text-opacity))",
  },
  "violet": {
    "100": "rgb(var(--tw-color-violet-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-violet-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-violet-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-violet-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-violet-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-violet-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-violet-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-violet-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-violet-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-violet-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-violet-950)/var(--tw-text-opacity))",
  },
  "warmGray": {
    "100": "rgb(var(--tw-color-warmGray-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-warmGray-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-warmGray-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-warmGray-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-warmGray-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-warmGray-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-warmGray-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-warmGray-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-warmGray-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-warmGray-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-warmGray-950)/var(--tw-text-opacity))",
  },
  "white": "rgb(var(--tw-color-white)/var(--tw-text-opacity))",
  "yellow": {
    "100": "rgb(var(--tw-color-yellow-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-yellow-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-yellow-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-yellow-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-yellow-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-yellow-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-yellow-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-yellow-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-yellow-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-yellow-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-yellow-950)/var(--tw-text-opacity))",
  },
  "zinc": {
    "100": "rgb(var(--tw-color-zinc-100)/var(--tw-text-opacity))",
    "200": "rgb(var(--tw-color-zinc-200)/var(--tw-text-opacity))",
    "300": "rgb(var(--tw-color-zinc-300)/var(--tw-text-opacity))",
    "400": "rgb(var(--tw-color-zinc-400)/var(--tw-text-opacity))",
    "50": "rgb(var(--tw-color-zinc-50)/var(--tw-text-opacity))",
    "500": "rgb(var(--tw-color-zinc-500)/var(--tw-text-opacity))",
    "600": "rgb(var(--tw-color-zinc-600)/var(--tw-text-opacity))",
    "700": "rgb(var(--tw-color-zinc-700)/var(--tw-text-opacity))",
    "800": "rgb(var(--tw-color-zinc-800)/var(--tw-text-opacity))",
    "900": "rgb(var(--tw-color-zinc-900)/var(--tw-text-opacity))",
    "950": "rgb(var(--tw-color-zinc-950)/var(--tw-text-opacity))",
  },
}
`;
